File: src/main/java/ru/ifmo/is/lab1/dragons/dto/DragonUpdateDto.java
package ru.ifmo.is.lab1.dragons.dto;

import lombok.Data;
import jakarta.validation.constraints.*;
import org.openapitools.jackson.nullable.JsonNullable;
import ru.ifmo.is.lab1.dragons.DragonType;

@Data
public class DragonUpdateDto {
  @NotNull
  @NotBlank
  private JsonNullable<String> name;

  @NotNull
  private JsonNullable<Integer> coordinatesId;

  private JsonNullable<Integer> caveId;

  private JsonNullable<Integer> killerId;

  @NotNull
  private JsonNullable<Integer> headId;

  @NotNull
  private JsonNullable<DragonType> type;

  @Positive
  private JsonNullable<Integer> age;

  @Positive
  private JsonNullable<Integer> wingspan;

  private JsonNullable<Boolean> speaking;
}

File: frontend/src/interfaces/dto/dragons/DragonUpdateDto.ts
import { DragonType } from '../../models/DragonType';
import { JsonNullable } from '../../models/JsonNullable';

export interface DragonUpdateDto {
    name?: JsonNullable<string>;
    coordinatesId?: JsonNullable<number>;
    caveId?: JsonNullable<number>;
    killerId?: JsonNullable<number>;
    headId?: JsonNullable<number>;
    type?: JsonNullable<DragonType>;
    age?: JsonNullable<number>;
    wingspan?: JsonNullable<number>;
    speaking?: JsonNullable<boolean>;
}
